aws/eks/eks-how-to/deploy-ubuntu-pro-cluster.rst

```diff
@@ -0,0 +1,282 @@
+custom Ubuntu Pro AMI (with `Packer <https://www.packer.io/>`_) and use it at cluster creation time.
+
+ 
+What are the caveats?
+*********************
+
+You need to keep rebuilding your custom Ubuntu Pro image in order to have the
+latest updates and security fixes from upstream.
+
+Also, storing AMI has a cost on AWS, and you will have to replicate it to multiple regions if you need to.
+
+Get and modify the Packer file
+******************************
+
+Create an ``eks-ubuntu-fips.json`` file with the following (replacing the
+placeholder credentials with your own in the "variables" section):
+
+..  code-block:: bash
+
+    {
+        "variables": {
+            "aws_access_key": "YOUR_IAM_ACCESS_KEY",
+            "aws_secret_key": "YOUR_IAM_SECRET_KEY",
+            "ua_token": "YOUR_UA_TOKEN",
+            "eks_ver": "1.23"
+        },
+        "builders": [
+        {
+            "type": "amazon-ebs",
+            "access_key": "{{user `aws_access_key`}}",
+            "secret_key": "{{user `aws_secret_key`}}",
+            "region": "us-east-1",
+            "instance_type": "t2.micro",
+            "ami_name": "eks{{user `eks_ver`}}-fips-ubuntu20.04-{{timestamp}}",
+            "source_ami_filter": {
+                "filters": {
+                    "virtualization-type": "hvm",
+                    "name": "ubuntu-eks/k8s_{{user `eks_ver`}}/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*",
+                    "root-device-type": "ebs"
+                },
+            "owners": ["099720109477"],
+            "most_recent": true
+        },
+        "ssh_username": "ubuntu"
+        }
+        ],
+        "provisioners": [
+        {
+            "type": "shell",
+            "inline": [
+            "cloud-init status --wait",
+            "sudo apt-get update && sudo apt-get upgrade -y --with-new-pkgs"
+            ]
+        },
+        {
+            "type": "shell",
+            "inline": [
+            "sudo ua attach {{user `ua_token`}}",
+            "sudo ua status --wait",
+            "sudo ua enable fips --assume-yes"
+            ]
+        },
+        {
+            "type": "shell",
+            "inline": [
+            "sudo truncate -s 0 /etc/machine-id",
+            "sudo truncate -s 0 /var/lib/dbus/machine-id"
+            ]
+        }
+        ]
+    }
+
+This is the file that will be used by Packer to build the custom Ubuntu Pro AMI.
+
+Remember that the final AMI needs to be in the same region as the EKS cluster, 
+so make sure to adjust the "region" above accordingly.
+
+This Packer file takes as a source an existing AMI of an EKS-based Ubuntu Focal
+Server for amd64. It will then launch shell commands to wait for cloud-init to
+finish and upgrade the system. Afterwards, it attaches the machine to a Pro subscription
+using your UA token and enables FIPS. To conclude, it removes the machine-id
+from the custom image, to have a unique machine-id on every node instantiation.
```

Comment:
typo - installation

Reply:
I think instantiation is a good, because each instance will have an unique machine-id on launch